### PR TITLE
feat: add view all surveys page

### PR DESCRIPTION
**Pull request by muskankhedia**
## Description
Added the view survey page option with the display of the complete list of all the surveys taken by the client.

## Related issues and discussion
#617 

## Screenshots, if any
screencapture-localhost-4200-clients-1-actions-Survey-2020-07-13-12_57_19

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] If you have multiple commits please combine them into one commit by squashing them.

- [X] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by muskankhedia** [comment]
@abhaychawla Please review the PR and let me know if any changes to be made.

File: src/app/clients/clients-view/view-survey/view-survey.component.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<mat-card>
+  <mat-card-title>List of All Surveys</mat-card-title>
```
Comment (karantakalkar) [inline src/app/clients/clients-view/view-survey/view-survey.component.html:2]:
```suggestion
```

File: src/app/clients/clients-view/view-survey/view-survey.component.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+          <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+          <tr mat-row *matRowDef="let row; columns: displayedColumns;" class="select-row"></tr>
+
+        </table>
+
+        <mat-paginator [pageSizeOptions]="[10, 25, 50, 100]" showFirstLastButtons></mat-paginator>
+
+      </div>
+    </div>
```
Comment (karantakalkar) [inline src/app/clients/clients-view/view-survey/view-survey.component.html:52]: spacing

File: src/app/clients/clients-view/view-survey/view-survey.component.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  </mat-card-content>
+
+</mat-card>
```
Comment (karantakalkar) [inline src/app/clients/clients-view/view-survey/view-survey.component.html:55]: eof

File: src/app/clients/clients-view/view-survey/view-survey.component.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  selector: 'mifosx-view-survey',
+  templateUrl: './view-survey.component.html'
+})
+export class ViewSurveyComponent implements OnInit {
+
+  surveyData: any;
+  /** Data source for view surveys table. */
+  dataSource: MatTableDataSource<any>;
+  /** Columns to be displayed in list of surveys table. */
+  displayedColumns: string[] = ['name', 'createdBy', 'date', 'score'];
```
Comment (karantakalkar) [inline src/app/clients/clients-view/view-survey/view-survey.component.ts:15]: 1 line space b/w variables and viewchilds

File: src/app/clients/clients-view/view-survey/view-survey.component.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    data.forEach((ele: { id: number, userId: number, username: string, clientId: number, surveyId: number, surveyName: string, scorecardValues: Object[] }) => {
+      ele.scorecardValues.forEach((element: { createdOn: number, value: number }) => {
+        const item = {} as { surveyName: string, createdby: string, date: number, score: number };
+        item.surveyName = ele.surveyName;
+        item.createdby = ele.username;
+        item.date = element.createdOn;
+        item.score = element.value;
+        surveys.push(item);
+      });
+    });
```
Comment (karantakalkar) [inline src/app/clients/clients-view/view-survey/view-survey.component.ts:47]: no need for this function
```suggestion
```

File: src/app/clients/clients-view/view-survey/view-survey.component.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+    this.dataSource = new MatTableDataSource(surveys);
```
Comment (karantakalkar) [inline src/app/clients/clients-view/view-survey/view-survey.component.ts:49]:
```suggestion
    this.dataSource = new MatTableDataSource(data);
```

File: src/app/clients/clients-view/view-survey/view-survey.component.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    this.route.data.subscribe((data: { survey: any }) => {
+      this.surveyData = data.survey;
+    });
+  }
+
+  ngOnInit() {
+    this.constructSurvey(this.surveyData);
+  }
+
+  constructSurvey(data: any) {
```
Comment (karantakalkar) [inline src/app/clients/clients-view/view-survey/view-survey.component.ts:36]:
```suggestion
  setSurveys(data: any) {
```

File: src/app/clients/clients-view/view-survey/view-survey.component.ts (hunk 4 above)
Comment (karantakalkar) [inline src/app/clients/clients-view/view-survey/view-survey.component.ts:15]:
```suggestion
  displayedColumns: string[] = ['surveyName', 'createdBy', 'date', 'score'];
```

File: src/app/clients/clients-view/view-survey/view-survey.component.ts [hunk 8]
```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
```
Comment (karantakalkar) [inline src/app/clients/clients-view/view-survey/view-survey.component.ts:1]: add relevant comments in entire file

File: src/app/clients/clients-view/view-survey/view-survey.component.html (hunk 1 above)
Comment (karantakalkar) [reply src/app/clients/clients-view/view-survey/view-survey.component.html:2]: refactor html for spacing

**Pull request merged**